### PR TITLE
MiKo_1049 is now able to fix some plural texts

### PR DESCRIPTION
**Pull request by RalfKoban**
- Improve requirement term analyzer to handle "ShouldBe" patterns and pluralization in method names

- Improve performance by adding thread-safe caching for name analysis

- Improve accuracy by deduplicating replacement rules, and fixing edge cases in term replacements

- Add tests for PascalCase and snake_case scenarios

### CODE REVIEW

File: MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs [hunk 1]
```diff
@@ -84,6 +84,12 @@ private static IEnumerable<Pair> CreateReplacementMapEntries()
                 yield return new Pair("_" + lowerTerm + "_handle_", "_handles_");
                 yield return new Pair("_" + lowerTerm + "_", "_does_");
             }
+
+            yield return new Pair("sIs", "sAre");
+            yield return new Pair("ssAre", "ssIs");
+
+            yield return new Pair("s_is_", "s_are_");
+            yield return new Pair("ss_are_", "ss_is_");
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs:113]: fixed

File: MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs [hunk 2]
```diff
@@ -159,6 +159,11 @@ public void DoSomething()
         [TestCase("Something_ShouldDocumentNothingForReasons", "Something_DoesDocumentNothingForReasons")]
         [TestCase("Something_ShouldHandleStuff", "Something_HandlesStuff")]
         [TestCase("Something_Should_HandleStuff", "Something_HandlesStuff")]
+        [TestCase("AppsShouldBeHandled", "AppsAreHandled")]
+        [TestCase("AppShouldBeHandled", "AppIsHandled")]
+        [TestCase("AccessShouldBeHandled", "AccessIsHandled")]
+        [TestCase("Apps_should_be_handled", "Apps_are_handled")]
+        [TestCase("Access_should_be_handled", "Access_is_handled")]
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs:177]: fixed

File: MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs [hunk 3]
```diff
@@ -12,7 +13,11 @@ public sealed class MiKo_1049_RequirementTermAnalyzer : NamingAnalyzer
     {
         public const string Id = "MiKo_1049";
 
-        private static readonly Pair[] ReplacementMap = CreateReplacementMapEntries().ToArray();
+        private static readonly Pair[] ReplacementMap = CreateReplacementMapEntries().Distinct().ToArray();
+
+        private static readonly ConcurrentDictionary<string, string> AnalyzeNameCache = new ConcurrentDictionary<string, string>();
+
+        private static readonly ConcurrentDictionary<string, string> BetterNameCache = new ConcurrentDictionary<string, string>();
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs]: fixed

File: MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs [hunk 4]
```diff
@@ -163,6 +165,21 @@ public void Code_gets_fixed_for_method_(string method, string wanted) => VerifyC
...
+        [TestCase("AppShouldBeHandled", "AppIsHandled")]
+        [TestCase("AccessShouldBeHandled", "AccessIsHandled")]
+        [TestCase("Apps_should_be_handled", "Apps_are_handled")]
+        [TestCase("Access_should_be_handled", "Access_is_handled")]
+        [TestCase("BusShouldBeLate", "BusIsLate")]
+        [TestCase("Bus_should_be_late", "Bus_is_late")]
+        [TestCase("TestRunsShouldBeOK", "TestRunsAreOK")]
+        [TestCase("Test_runs_should_Be_OK", "Test_runs_are_OK")]
+        [TestCase("ThisShouldBeOK", "ThisIsOK")]
+        [TestCase("This_should_be_OK", "This_is_OK")]
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs:190]: added

File: MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs [hunk 5]
```diff
@@ -63,44 +68,59 @@ private static IEnumerable<Pair> CreateReplacementMapEntries()
...
+                yield return new Pair("_" + lowerTerm + "_runss", "_runs"); // in-place fix for '_runs' just being replaced to '_runss'
                 yield return new Pair("_" + lowerTerm + "_throw_", "_throws_");
                 yield return new Pair("_" + lowerTerm + "_handle_", "_handles_");
                 yield return new Pair("_" + lowerTerm + "_", "_does_");
             }
+
+            yield return new Pair("sIs", "sAre");
+            yield return new Pair("isAre", "isIs");
+            yield return new Pair("ssAre", "ssIs");
+            yield return new Pair("usAre", "usIs");
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs:105]: added

File: MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs [hunk 6]
```diff
@@ -163,6 +165,23 @@ public void Code_gets_fixed_for_method_(string method, string wanted) => VerifyC
...
+        [TestCase("Apps_should_be_handled", "Apps_are_handled")]
+        [TestCase("Access_should_be_handled", "Access_is_handled")]
+        [TestCase("BusShouldBeLate", "BusIsLate")]
+        [TestCase("Bus_should_be_late", "Bus_is_late")]
+        [TestCase("GasShouldBeOn", "GasIsOn")]
+        [TestCase("Gas_should_be_on", "Gas_is_on")]
+        [TestCase("TestRunsShouldBeOK", "TestRunsAreOK")]
+        [TestCase("Test_runs_should_Be_OK", "Test_runs_are_OK")]
+        [TestCase("ThisShouldBeOK", "ThisIsOK")]
+        [TestCase("This_should_be_OK", "This_is_OK")]
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Tests/Rules/Naming/MiKo_1049_RequirementTermAnalyzerTests.cs:190]: tests added

File: MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs [hunk 7]
```diff
@@ -63,44 +68,61 @@ private static IEnumerable<Pair> CreateReplacementMapEntries()
...
                 yield return new Pair("_" + lowerTerm + "_handle_", "_handles_");
                 yield return new Pair("_" + lowerTerm + "_", "_does_");
             }
+
+            yield return new Pair("sIs", "sAre");
+            yield return new Pair("asAre", "asIs");
+            yield return new Pair("isAre", "isIs");
+            yield return new Pair("ssAre", "ssIs");
+            yield return new Pair("usAre", "usIs");
+
```
[the remark this answers is not in the document]
Comment (RalfKoban) [reply MiKo.Analyzer.Shared/Rules/Naming/MiKo_1049_RequirementTermAnalyzer.cs:106]: fixups added
